File: 921MinimumAddToMakeParenthesesValid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        vector<char> arr;
        for (int i = 0 ; i < s.length() ; i++){
            if (arr.size() > 0 && arr[arr.size() - 1] == '(' && s[i] == ')'){
                arr.pop_back();
            }
            else{
                arr.push_back(s[i]);
            }
        }
        return arr.size();
    }
};

File: 921MinimumAddToMakeParenthesesValid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        arr = []
        for i in s:
            if len(arr) != 0:
                if arr[-1] == "(" and i == ")":
                    arr.pop(-1)
                else:
                    arr.append(i)
            else:
                arr.append(i)
        return len(arr)
